src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,198 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final int OFFSET_ONE = 1;
+    private static final int EMPTY_LIST = 0;
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
+    private static class Node<E> {
+        private E item;
+        private Node<E> next;
+        private Node<E> prev;
+
+        public Node(Node<E> prev, E item, Node<E> next) {
+            this.item = item;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
     @Override
     public void add(T value) {
+        linkLast(value);
     }
 
     @Override
     public void add(T value, int index) {
+        checkInsertIndex(index);
+        if (index == size) {
+            linkLast(value);
+            return;
+        }
+        Node<T> newNode = new Node<>(null, value, null);
+        if (index == EMPTY_LIST) {
+            newNode.next = first;
+            first = newNode;
+            size++;
+            return;
+        }
+        Node<T> temp = first;
+        for (int i = 0; i < index - OFFSET_ONE; i++) {
+            temp = temp.next;
+        }
+        temp.next.prev = newNode;
+        newNode.next = temp.next;
+        newNode.prev = temp;
+        temp.next = newNode;
```

Comment:
Don't create repeating code. You should consider creating a `findNodeByIndex` method to avoid code duplication for finding a node at a certain index. This logic is repeated in other methods as well.